fedimint-server/src/consensus/mod.rs

```diff
@@ -153,14 +162,19 @@ impl FedimintConsensus {
         db: Database,
         module_inits: ModuleGenRegistry,
         modules: ModuleRegistry<DynServerModule>,
-    ) -> Self {
-        Self {
-            modules,
-            cfg,
-            module_inits,
-            db,
-            transaction_notify: Arc::new(Notify::new()),
-        }
+    ) -> (Self, Receiver<Transaction>) {
+        let (tx_sender, tx_receiver) = mpsc::channel(TRANSACTION_BUFFER_SIZE);
+
+        (
+            Self {
+                modules,
+                cfg,
+                module_inits,
+                db,
+                tx_sender,
+            },
+            tx_receiver,
+        )
```

Comment:
I feel like we need to overthink the `FedimintConsensus` and `FedimintServer` border. Returning the receiver separately is very weird. No blocker for this PR though.

Reply:
Agreed, could not put receiver in `FedimintConsensus` without wrapping it in a `Mutex` which seems messier.  Right now feels like it could be broken into an interface between the API thread / consensus thread AND the outcome processing code.

Reply:
See #172